pkg/analysis/markerscope/schema.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package markerscope
+
+import "go/types"
+
+// SchemaType represents OpenAPI schema types that markers can target.
+type SchemaType string
+
+const (
+	// SchemaTypeInteger represents integer types (int, int32, int64, uint, etc.)
+	SchemaTypeInteger SchemaType = "integer"
+	// SchemaTypeNumber represents floating-point types (float32, float64).
+	SchemaTypeNumber SchemaType = "number"
+	// SchemaTypeString represents string types.
+	SchemaTypeString SchemaType = "string"
+	// SchemaTypeBoolean represents boolean types.
+	SchemaTypeBoolean SchemaType = "boolean"
+	// SchemaTypeArray represents array/slice types.
+	SchemaTypeArray SchemaType = "array"
+	// SchemaTypeObject represents struct/map types.
+	SchemaTypeObject SchemaType = "object"
+)
```

Comment:
Does this all need to be exported?

Reply:
This is configured this way because it's necessary for the Configuration settings.